GrappleGirlGame/Music.lua

```diff
@@ -0,0 +1,52 @@
+--[[
+Music = {}
+Music.queue = {}
+
+function Music:play(data)
+    -- In love2d you can use Static or Stream for playing Music
+	local src = love.audio.newSource(data, "static")
+    src:setVolume(0.5)
+    src:play()
+end
+--]]
+do
+    -- will hold the currently playing sources
+    local sources = {}
+ 
+    -- check for sources that finished playing and remove them
+    -- add to love.update
+    function love.audio.update()
+        local remove = {}
+        for _,s in pairs(sources) do
+            if s:isStopped() then
+                remove[#remove + 1] = s
+            end
+        end
+ 
+        for i,s in ipairs(remove) do
+            sources[s] = nil
+        end
+    end
+ 
+    -- overwrite love.audio.play to create and register source if needed
+    local play = love.audio.play
+    function love.audio.play(what, how, loop)
```

Comment:
Rather than overriding love.audio.play and love.audio.stop, you should create a global table called Music or something with its own methods that use your code. While it's probably fine to redefine functions provided by love2d *in this situation*, it's still something I would like to avoid, except for the ones you are explicitly supposed to, like love.update (and for this, love.audio.update is also fine).